apps/sync/js/adapters/bookmarks.js

```diff
@@ -167,12 +186,66 @@ var BookmarksHelper = (() => {
     });
   }
 
+  function checkIfCleared(userid) {
+    return getLastRevisionId(userid).then(lastRevisionId => {
+      return _ensureStore().then(store => {
+        if (lastRevisionId === null) {
+          // Skip first task which is always { id: null, operation: 'clear' }
+          var cursor = store.sync();
+          cursor.next().then(() => {
+            return cursor;
+          });
+        }
+        return store.sync(lastRevisionId);
+      });
+    }).then(cursor => {
+      var hasCleared = false;
+      return new Promise(resolve => {
+        function runNextTask(cursor) {
+          cursor.next().then(task => {
+            if (task.operation === 'done') {
+              resolve({
+                revisionId: task.revisionId,
+                hasCleared
+              });
+            } else {
+              if (task.operation === 'clear') {
+                hasCleared = true;
+              }
+              // Avoid stack overflow:
+              setTimeout(() => {
+                // Will eventually get to a 'done' task:
+                runNextTask(cursor);
+              });
+            }
+          });
+        }
+        runNextTask(cursor);
+      });
+    });
+  }
+
+  function handleClear(userid) {
+    var lastRevisionId;
+    return checkIfCleared(userid)
```

Comment:
https://github.com/mozilla-b2g/gaia/pull/32619/files#r43370562

I think it would be more clear if we use `getLastRevisionId` here.